sstable/writer.go

```diff
@@ -269,6 +273,11 @@ func (w *Writer) addPoint(key InternalKey, value []byte) error {
 		// semantically identical, because we need to ensure that SmallestPoint.UserKey
 		// is not nil. This is required by WriterMetadata.Smallest in order to
 		// distinguish between an unset SmallestPoint and a zero-length one.
+		// NB: We don't close this off of some alloc-pool since it will outlive this
+		// Writer as a field of the returned metadata that users like compaction.go
+		// then hang on to, and would otherwise continue to alias that whole pool's
+		// slice if we did so. So since we'll need to allocate it its own slice at
+		// some point anyway, we may as well do so here.
 		w.meta.SmallestPoint = w.meta.LargestPoint.Clone()
```

Comment:
This is the only diff from the reverted commit: originally it replaced this Clone with one from the pooled alloc, but here is left as-is, and the the comment is added to explain why it should stay that way.